src/content/en/showcase/2016/service-worker-perf.md

```diff
@@ -0,0 +1,679 @@
+
+**1. Track more events related to the load experience**
+
+We tracked several events that correspond to a technical metric (e.g. [HTMLImportsLoaded](http://webcomponents.org/polyfills/html-imports/), [WebComponentsReady](https://github.com/webcomponents/webcomponentsjs#webcomponentsready), etc.), but since so much of the load was done asynchronously, the point at which these events fired didn't necessarily correspond with a particular moment in the overall load experience.
+
+The primary load-related event we didn't track (but wish we had) is the point at which the splash screen disappeared and the user could see the page content.
+
+**2. Store the analytics client ID in IndexedDB**
+
+By default, analytics.js stores the [client ID field](/analytics/devguides/collection/analyticsjs/cookies-user-id) in the browser's cookies; unfortunately, service worker scripts cannot access cookies.
+
+This presented a problem for us when we tried to implement notification tracking. We wanted to send an event from the service worker (via the [Measurement Protocol](/analytics/devguides/collection/protocol/)) each time a notification was sent to a user, and then track the re-engagement success of that notification if the user clicked on it and arrived back in the app.
+
+While we were able to track the success of notifications in general via the `utm_source` [campaign parameter](https://support.google.com/analytics/answer/1033863#parameters), we weren't able to tie a particular re-engagement session to a particular user.
+
+What we could have done to workaround this limitation was store the client ID via IndexedDB in our tracking code, and then that value would have been accessible to the service worker script.
+
+**3. Let the service worker report online/offline status**
+
+Inspecting `navigator.onLine` will let you know if your browser is able to connect to the router or local area network, but it won't necessarily tell if you if the user has real connectivety. And since our offline analytics service worker script simply replayed failed hits (without modifying them, or marking them as failed), we were probably under-reporting our offline usage.
+
+In the future we should track both the status of `navigator.onLine` as well as whether the hit was replayed by the service worker due to an initial network failure. This will give us a more accurate picture of the true offline usage.
+
+## Wrapping up
+
+This case study has shown that using service worker did indeed improve the load performance of the Google I/O webapp across a wide range of browsers, networks, and devices. It has also shown that when you look at a distribution of the load data across a wide range of browsers, networks, and devices, you get much more insight into how this technology handles in real-world scenarios, and you discover performance characteristics that you may not have expected.
+
+Here were some of the key takeaways from the IOWA study:
+
```

Comment:
I actually disagree here. This case study isn't about what the results were, it's about how the results were measured. If someone just read these bullet points, they would completely miss the point of the article.

I'm not opposed to adding a tl;dr, I just don't know what else I'd say that's not already in the intro.